### ISSUE TITLE
should Digest implement Hash?

### ISSUE THREAD

**Issue by oconnor663** [oconnor663/blake2_c.rs#9]

**Comment by asomers** [comment]
No, I don't think so.  `Digest` is the output of a hash function, but the `Hash` trait is for the input to hash functions.  Instead, I think `blake2` should contain a struct that implements `Hasher`

**Comment by oconnor663** [comment]
Now that I look it up, it seems like `ArrayVec` implements `Hash` itself, so at least callers can use the exposed `bytes` field if they need something to put in a `HashMap`.
